code/modules/asset_cache/spritesheet/legacy/legacy_spritesheet.dm

```diff
@@ -0,0 +1,418 @@
+
+	return TRUE
+
+/datum/asset/spritesheet/proc/read_data_from_cache()
+	var/json = json_decode(rustg_file_read(data_cache_filename()))
+
+	if (islist(json["sprites"]))
+		sprites = json["sprites"]
+
+	return TRUE
+
+/datum/asset/spritesheet/proc/send_from_cache(client/client)
+	if (isnull(cached_spritesheets_needed))
+		stack_trace("cached_spritesheets_needed was null when sending assets from [type] from cache")
+		cached_spritesheets_needed = list()
+
+	return SSassets.transport.send_assets(client, cached_spritesheets_needed + "spritesheet_[name].css")
+
+/// Returns the URL to put in the background:url of the CSS asset
+/datum/asset/spritesheet/proc/get_background_url(asset)
+	if (generating_cache)
+		return "%[asset]%"
+	else
+		return SSassets.transport.get_asset_url(asset)
+
+/datum/asset/spritesheet/proc/write_to_cache()
+	write_css_to_cache()
+	write_data_to_cache()
+
+/datum/asset/spritesheet/proc/write_css_to_cache()
+	for (var/size_id in sizes)
+		fcopy(SSassets.cache["[name]_[size_id].png"].resource, "[ASSET_CROSS_ROUND_CACHE_DIRECTORY]/spritesheet.[name]_[size_id].png")
+
+	generating_cache = TRUE
+	var/mock_css = generate_css()
+	generating_cache = FALSE
+
+	rustg_file_write(mock_css, css_cache_filename())
+
+/datum/asset/spritesheet/proc/write_data_to_cache()
+	rustg_file_write(json_encode(list(
+		"sprites" = sprites,
+	)), data_cache_filename())
+
+/datum/asset/spritesheet/proc/get_cached_url_mappings()
+	var/list/mappings = list()
+	mappings["spritesheet_[name].css"] = SSassets.transport.get_asset_url("spritesheet_[name].css")
+
+	for (var/asset_name in cached_spritesheets_needed)
+		mappings[asset_name] = SSassets.transport.get_asset_url(asset_name)
+
+	return mappings
+
+/// Override this in order to start the creation of the spritehseet.
+/// This is where all your Insert, InsertAll, etc calls should be inside.
+/datum/asset/spritesheet/proc/create_spritesheets()
+	SHOULD_CALL_PARENT(FALSE)
+	CRASH("create_spritesheets() not implemented for [type]!")
+
+/datum/asset/spritesheet/proc/Insert(sprite_name, icon/I, icon_state="", dir=SOUTH, frame=1, moving=FALSE)
+	if(should_load_immediately())
+		queuedInsert(sprite_name, I, icon_state, dir, frame, moving)
+	else
+		to_generate += list(args.Copy())
+
+/datum/asset/spritesheet/proc/queuedInsert(sprite_name, icon/I, icon_state="", dir=SOUTH, frame=1, moving=FALSE)
+#ifdef UNIT_TESTS
+	if (I && icon_state && !icon_exists(I, icon_state)) // check the base icon prior to extracting the state we want
+		stack_trace("Tried to insert nonexistent icon_state '[icon_state]' from [I] into spritesheet [name] ([type])")
+		return
+#endif
+	I = icon(I, icon_state=icon_state, dir=dir, frame=frame, moving=moving)
+	if (!I || !length(icon_states(I)))  // that direction or state doesn't exist
+		return
+
+	var/start_usage = world.tick_usage
+
+	//any sprite modifications we want to do (aka, coloring a greyscaled asset)
+	I = ModifyInserted(I)
+	var/size_id = "[I.Width()]x[I.Height()]"
```

Comment:
`get_icon_dimensions`? (I'll stop pointing out this and the icon variables called capital I for now)